records/2023-11-24.system-status.md

```diff
@@ -0,0 +1,162 @@
+| API         | DOWN     | The API is not responding                                                              |
+|             |          |                                                                                        |
+| File upload | UP       | The DD-API is responding to requests in 400ms or less                                  |
+| File upload | DEGRADED | The DD-API is responding to requests in more than 400ms                                |
+| File upload | DOWN     | The DD- API is not responding                                                          |
+|             |          |                                                                                        |
+| Email       | UP       | Priority emails are sending within 60s, and Other emails are sending within 45m        |
+| Email       | DEGRADED | Priority emails are taking longer than 60s, or Other emails are taking longer than 45m |
+| Email       | DOWN     | No emails sent within the last 5 minutes                                               |
+|             |          |                                                                                        |
+| SMS         | UP       | Priority SMSs are sending within 60s, and Other SMSs are sending within 45m            |
+| SMS         | DEGRADED | Priority SMSs are taking longer than 60s, or Other SMSs are taking longer than 45m     |
+| SMS         | DOWN     | No SMS messages sent within the last 5 minutes                                         |
+
+## Monitoring
+Each system will need to be monitored every 5m to determine which status it is in.
+
+## Systems architecture
+The website that shows system status needs to be deployed separately from the main GC Notify website so that we can continue showing status information even if the GC Notify website is down.  It should not live within our existing EKS node.
+
+### Front-end
+A new front-end website will be deployed to show the status information.  It will be served from a new domain, status.notification.canada.ca
+
+### Back-end
+A new back-end API wil be deployed that will be responsible for assessing the status of each component of the system.
+
+### Deployment
+
+#### Api Code
+We have two options to deploy the API's associated with the system architecture page.
+1. API-lambda
+    We can change the current heartbeat API lambda to make all the API calls required for the system status page.
+1. ECS Cluster
+    We can set up an ECS task (similar to Blazer/Perf-tests) to allow all the API calls
+
+Given we use both versions in our current code base and neither has any significant advantage over the other, we are going to use the API lambda (Option 1)
+
+The API lambda needs to accomplish the following things:
+1. Call the different APIs to get the information to display on the status page
+1. Call the Articles endpoint to get information around Incidents
```

Comment:
@jumana, I think we could just do this from the front-end instead, and not have to copy/relay the content.